analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -77,13 +78,16 @@ class Analyzer(private val config: AnalyzerConfiguration, private val labels: Ma
             Pair(manager, mappedFiles).takeIf { mappedFiles.isNotEmpty() }
         }.toMap(mutableMapOf())
 
-        val hasDefinitionFileInRootDirectory = managedFiles.values.flatten().any {
-            it.parentFile.absoluteFile == absoluteProjectPath
+        // Check whether there are unmanaged files (because of deactivated, unsupported, or non-present package
+        // managers) which we need to attach to an artificial "unmanaged" project.
+        val managedDirs = managedFiles.values.flatten().mapNotNull { it.parentFile }
+        val hasOnlyManagedDirs = absoluteProjectPath in managedDirs || absoluteProjectPath.listFiles().orEmpty().all {
+            it in managedDirs || it.name in VCS_DIRECTORIES
         }
 
-        val unmanagedFactory = packageManagers.find { it is Unmanaged.Factory }
-        if (unmanagedFactory != null && (factoryFiles.isEmpty() || !hasDefinitionFileInRootDirectory)) {
-            unmanagedFactory.create(absoluteProjectPath, config, repositoryConfiguration).let {
+        if (!hasOnlyManagedDirs) {
+            val unmanagedFactory = packageManagers.find { it is Unmanaged.Factory }
+            unmanagedFactory?.create(absoluteProjectPath, config, repositoryConfiguration)?.also {
```

Comment:
With the simplified condition this could now be compressed a bit, for example:
```kotlin
        packageManagers.find { it is Unmanaged.Factory }
            ?.create(absoluteProjectPath, config, repositoryConfiguration)
            ?.run { managedFiles[this] = listOf(absoluteProjectPath) }
```